lib/actions/apiV2.js

```diff
@@ -0,0 +1,533 @@
+          dispatch(setMapZoom(10))
+        }
+      },
+      rewritePayload: (payload) => {
+        const { stop } = payload?.data
+        if (!stop) return findStopError()
+
+        const color = stop.routes?.length > 0 && `#${stop.routes[0].color}`
+
+        // Doing some OTP1 compatibility rewriting here
+        return {
+          ...stop,
+          agencyId: stop?.route?.agency?.gtfsId,
+          agencyName: stop?.route?.agency?.name,
+          color
+        }
+      },
+      serviceId: 'stops'
+    }
+  )
+}
+
+const getVehiclePositionsForRoute = () => {
+  return function (dispatch, getState) {
+    console.warn('OTP2 does not yet support vehicle positions for route!')
+  }
+}
+
+export function findRoute(params) {
+  return function (dispatch, getState) {
+    const { routeId } = params
+    if (!routeId) return
+
+    return dispatch(
+      createGraphQLQueryAction(
+        `{
+        route(id: "${routeId}") {
+          id: gtfsId
+          desc
+          agency {
+            id: gtfsId
+            name
+            url
+            timezone
+            lang
+            phone
+          }
+          longName
+          type
+          color
+          textColor
+          bikesAllowed
+          routeBikesAllowed: bikesAllowed
+      
+          patterns {
+            id
+            name
+            patternGeometry {
+              points
+              length
+            }
+            stops {
+              code
+              id: gtfsId
+              lat
+              lon
+              name
+              locationType
+              geometries {
+                geoJson
+              }
+              routes {
+                color
+              }
+            }
+          }
+        }
+      }
+      `,
+        {},
+        findRouteResponse,
+        findRouteError,
+        {
+          noThrottle: true,
+          // TODO: avoid re-writing OTP2 route object to match OTP1 style
+          rewritePayload: (payload) => {
+            if (payload.errors) {
+              return dispatch(findRouteError(payload.errors))
+            }
+            const { route } = payload?.data
+            if (!route) return
+
+            const routePatterns = {}
+            route.patterns.forEach((pattern) => {
+              const patternStops = pattern.stops.map((stop) => {
+                const color =
+                  stop.routes?.length > 0 && `#${stop.routes[0].color}`
+                if (stop.routes) delete stop.routes
+                return { ...stop, color }
+              })
+              routePatterns[pattern.id] = {
+                ...pattern,
+                desc: pattern.name,
+                geometry: pattern.patternGeometry,
+                stops: patternStops
+              }
+            })
+            route.patterns = routePatterns
+            // TODO: avoid explicit behavior shift like this
+            route.v2 = true
+
+            return route
```

Comment:
In that case it's better to create a shallow copy of `payload.data` that gets modified (like in the other methods above) or, perhaps more preferably, perform a `clone`.

Reply:
Good call thanks! Addressed in d2507aa982ecca23a82fb1964c5e7a7099b81d88